objectivec/cxx_utils.mm

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "cxx_utils.h"
+
+#import <vector>
+#import <optional>
+#import <string>
+
+#import "error_utils.h"
+
+#import "ort_value_internal.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace utils {
+
+NSString* toNSString(const std::string& str) {
+  NSString* nsStr = [NSString stringWithUTF8String:str.c_str()];
+  if (!nsStr) {
+    ORT_CXX_API_THROW("Failed to convert std::string to NSString", ORT_INVALID_ARGUMENT);
+  }
+
+  return nsStr;
+}
+
+NSString* _Nullable toNullableNSString(const std::optional<std::string>& str) {
+  if (str.has_value()) {
+    return [NSString stringWithUTF8String:str->c_str()];
```

Comment:
nit: can do the check here too, e.g., by calling the other helper.
```suggestion
    return toNSString(*str);
```

Reply:
wouldn't be checking the nullability of string twice (once with has.value() and other inside `toNSString`?

Reply:
I think the difference is that the `optional<std::string>` being empty is an expected case, whereas the NSString creation failure is not expected